src/tests/ftest/launch.py

```diff
@@ -51,2290 +66,2896 @@
-                # Simply add the new value found for this key
-                matches[found_key] = found[found_key]
+        # Create a new TestResult for this test
+        self.result.tests.append(TestResult(class_name, test_name, log_file, self.logdir))
 
-            else:
-                is_list = isinstance(matches[found_key], list)
-                if not is_list:
-                    matches[found_key] = [matches[found_key]]
-                if isinstance(found[found_key], list):
-                    for found_item in found[found_key]:
-                        if found_item not in matches[found_key]:
-                            matches[found_key].append(found_item)
-                elif found[found_key] not in matches[found_key]:
-                    matches[found_key].append(found[found_key])
+        # Mark the start of the processing of this test
+        self.result.tests[-1].start()
 
-                if not is_list and len(matches[found_key]) == 1:
-                    matches[found_key] = matches[found_key][0]
+        return self.result.tests[-1]
 
-    matches = {}
-    if isinstance(obj, val_type) and isinstance(key, str) and key in keys:
-        # Match found
-        matches[key] = obj
-    elif isinstance(obj, dict):
-        # Recursively look for matches in each dictionary entry
-        for obj_key, obj_val in list(obj.items()):
-            add_matches(find_values(obj_val, keys, obj_key, val_type))
-    elif isinstance(obj, list):
-        # Recursively look for matches in each list entry
-        for item in obj:
-            add_matches(find_values(item, keys, None, val_type))
+    def _end_test(self, test_result, message, fail_class=None, exc_info=None):
+        """Mark the end of the test result.
 
-    return matches
+        Args:
+            test_result (TestResult): the test result to complete
+            message (str): exit message or reason for failure
+            fail_class (str, optional): failure category.
+            exc_info (OptExcInfo, optional): return value from sys.exc_info().
+        """
+        if fail_class is None:
+            self._pass_test(test_result, message)
+        else:
+            self._fail_test(test_result, fail_class, message, exc_info)
+        if test_result:
+            test_result.end()
 
+    @staticmethod
+    def _pass_test(test_result, message=None):
+        """Set the test result as passed.
 
-def get_test_list(tags):
-    """Generate a list of tests and avocado tag filter from a list of tags.
+        Args:
+            test_result (TestResult): the test result to mark as passed
+            message (str, optional): explanation of test passing. Defaults to None.
+        """
+        if message is not None:
+            logger.debug(message)
+        test_result.status = TestResult.PASS
 
-    Args:
-        tags (list): a list of tag or test file names
+    def _fail_test(self, test_result, fail_class, fail_reason, exc_info=None):
```

Comment:
(pylint-no-self-use) Method could be a function

Reply:
This was not caught by pylint in https://github.com/daos-stack/daos/commit/49e743b057274805ca26cc9ee8b35de993681997